.github/workflows/register_repositories.py

```diff
@@ -0,0 +1,216 @@
+    url = f'{API_BASE_URL}/repos/{repo}/hooks'
+
+    async with aiohttp.ClientSession(headers=headers) as client:
+
+        logging.info('Retrieving repository webhooks metadata.')
+        async with client.get(url) as response:
+            hooks = await response.json()
+            if response.status == 200:
+                potl_hooks = []
+                for hook in hooks:
+                    if (
+                        set(hook['events']) == {'issues', 'pull_request'}
+                        and hook['config']['url'] == XDEVBOT_MAIN_ENDPOINT
+                        and hook['config']['content_type'] == 'json'
+                        and hook['active']
+                    ):
+                        potl_hooks.append(hook)
+
+                if len(potl_hooks) == 0:
+                    logging.info('Creating repository webhook.')
+                    request = dict(
+                        name='web',
+                        events=['issues', 'pull_request'],
+                        config=dict(url=XDEVBOT_MAIN_ENDPOINT, content_type='json'),
+                    )
+                    async with client.post(url, json=request) as response:
+                        main_hook = await response.json()
+                        if response.status != 201:
+                            logging.error('Failed to create repository webhook.')
+
+                elif len(potl_hooks) == 1:
+                    logging.info('Existing repository webhook found.')
+                    main_hook = potl_hooks[0]
+
+                else:
+                    timestamps = [
+                        datetime.strptime(hook['updated_at'], '%Y-%m-%dT%H:%M:%SZ')
+                        for hook in potl_hooks
+                    ]
+                    newest_timestamp, i_hook = max((t, i) for (i, t) in enumerate(timestamps))[1]
+                    logging.info(
+                        f'Found {len(potl_hooks)} potential webhooks on the repository, '
+                        f'choosing most recent webhook at {newest_timestamp}.'
+                    )
+                    main_hook = potl_hooks[i_hook]
+
+                hooks_info[repo] = main_hook
+
+            else:
+                logging.error('Could not retrieve repository metadata.')
+
+
+def format_repo_url(repo):
+    return f'[{repo}](https://github.com/{repo})'
+
+
+if __name__ == '__main__':
+
+    config_file = 'config.yaml'
+    new_config, old_config, repos = configure(config_file)
+    if new_config != old_config:
+        with open(config_file, 'w') as file_obj:
+            yaml.round_trip_dump(new_config, file_obj, indent=2, block_seq_indent=2)
+
+    added_hooks = {}
+    removed_hooks = {}
+    loop = asyncio.get_event_loop()
+    tasks = [loop.create_task(install_repo_webhook(repo, added_hooks)) for repo in repos['add']] + [
+        loop.create_task(delete_repo_webhook(repo, removed_hooks)) for repo in repos['remove']
+    ]
+    loop.run_until_complete(asyncio.gather(*tasks))
```

Comment:
Doing this asynchronously, could this create a problem is someone adds and then removes a repo?  For example, what might happen if someone does:

```
/add-repo campaign:core repo:xdev2
/remove-repo campaign:core repo:xdev2
```

Might it try to remove the repo first?